src/util/parented_ptr.h

```diff
@@ -26,14 +26,21 @@ class parented_ptr final {
     explicit parented_ptr(T* t) noexcept
             : m_ptr{t} {
     }
-
+// Only generate destructor if not empty, otherwise its empty but will
+// cause the parented_ptr to be not trivially destructible even though it could be.
+#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
     ~parented_ptr() noexcept {
         DEBUG_ASSERT(!m_ptr || static_cast<const QObject*>(m_ptr)->parent());
     }
-
-    // Delete copy constructor and copy assignment operator
+#else
+    // explicitly generate trivial destructor (since decltype(m_ptr) is not a class type)
+    ~parented_ptr() noexcept = default;
+#endif
```

Comment:
for my understanding '~parented_ptr() = default' and `~parented_ptr() {}` is the same. So I can imagine that this is not needed.

Reply:
No its not equivalent. One if an empty definition which means that its still user provided even if it does nothing, the other one is the default declared one which does not count as user-provided. 

See
https://en.cppreference.com/w/cpp/language/destructor#Trivial_destructor
https://compiler-explorer.com/z/cbG99jbzr